File: rc-application/src/main/java/com/transkript/reportcard/business/service/SequenceService.java
package com.transkript.reportcard.business.service;


import com.transkript.reportcard.api.dto.SequenceDto;
import com.transkript.reportcard.api.dto.response.EntityResponse;
import com.transkript.reportcard.data.entity.Sequence;

import java.util.List;

public interface SequenceService {
    EntityResponse addSequence(SequenceDto sequenceDto);

    List<SequenceDto> getSequences();

    SequenceDto getSequence(Long id);

    EntityResponse updateSequence(Long id, SequenceDto sequenceDto);

    String deleteSequence(Long id);

    Sequence getSequenceEntity(Long sequenceId);
}

File: rc-application/src/main/java/com/transkript/reportcard/business/service/interf/SequenceService.kt
package com.transkript.reportcard.business.service.interf

import com.transkript.reportcard.api.dto.SequenceDto
import com.transkript.reportcard.api.dto.response.EntityResponse
import com.transkript.reportcard.data.entity.Sequence

interface SequenceService {
    fun addSequence(sequenceDto: SequenceDto?): EntityResponse?
    val sequences: List<SequenceDto?>?
    fun getSequence(id: Long?): SequenceDto?
    fun updateSequence(id: Long?, sequenceDto: SequenceDto?): EntityResponse?
    fun deleteSequence(id: Long?): String?
    fun getSequenceEntity(sequenceId: Long?): Sequence?
}
